### PR TITLE
fix: looser `is_local_representation`

### PR DESCRIPTION
**Pull request by wischli**
# Description

Fixes issue which blocks FRAX <> USDC order fulfilments on Demo.

The current implementation of `is_local_representation` is too strict such that it throws an error if the checked `CurrencyId` has not configured a local representation. Instead, it should return `Ok(false)`. This function has not been tested directly. 

To the best of my knowledge, this does not affect production for now as it only affects checking currencies which don't have a configured local asset, e.g. FRAX.

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: libs/types/src/tokens.rs [hunk 1]
```diff
@@ -431,18 +431,16 @@ where
...
+		if let Some(local) = meta_variant.additional.local_representation {
+			frame_support::ensure!(
+				meta_local.decimals == meta_variant.decimals,
+				DispatchError::Other("Mismatching decimals")
+			);
 
-		Ok(self == &local)
+			Ok(self == &local.into())
+		} else {
+			Ok(false)
```
Comment (lemunozm) [inline libs/types/src/tokens.rs:442]: Nice found!

File: pallets/token-mux/src/tests.rs [hunk 2]
```diff
@@ -506,3 +506,61 @@ pub(crate) mod swaps {
...
+	use cfg_types::tokens::CurrencyId;
+	use sp_runtime::DispatchError;
+
+	use super::*;
+	use crate::mock::{new_test_ext_invalid_assets, MockRegistry, USDC_WRONG_DECIMALS};
+
+	#[test]
+	fn is_local_happy_paths() {
+		new_test_ext().execute_with(|| {
+			assert_eq!(<CurrencyId as HasLocalAssetRepresentation<MockRegistry>>::is_local_representation_of(&USDC_LOCAL, &USDC_1), Ok(true));
```
Comment (lemunozm) [inline pallets/token-mux/src/tests.rs:521]: Just curious,

```rust
CurrencyId::is_local_representation_of()
```

Does not compile? Does the compiler force you to specify the trait?

**Review by lemunozm — APPROVED** [review]
Thanks for researching this @wischli, and for adding related tests

**Review by mustermeiszer — APPROVED** [review]
Thanks for the hustle! Nice found!

**Review by mustermeiszer — APPROVED** [review]

File: pallets/token-mux/src/tests.rs (hunk 2 above)
Comment (wischli) [reply pallets/token-mux/src/tests.rs:521]: Unfortunately not, the `AssetInspect` parameter type cannot be inferred. Tried different variants but none of the shortcuts worked.

**Pull request merged**